src/sitemap/handler.js

```diff
@@ -393,8 +394,188 @@ export async function sitemapAuditRunner(baseURL, context) {
+    auditData.auditResult.reasons.forEach((reason) => {
+      // sitemaps were found by no valid pages extracted
+      if (reason.error === ERROR_CODES.NO_SITEMAP_IN_ROBOTS) {
+        response.push({
+          getData: () => [{
+            sourceSitemapUrl: '',
+            pageUrl: `${auditUrl}/robots.txt`,
+            statusCode: 404,
+          }],
+          type: 'sitemap-not-available',
+          buildKey: (issue) => `${auditData.siteId}|sitemap-not-available|${issue.pageUrl}`,
+          title: 'No valid sitemap found in robots.txt',
+          description: 'Robots.txt does not contain a sitemap reference, fallback could not find useable sitemap',
+        });
+      }
+
+      // sitemaps were found by no valid pages extracted
+      if (reason.error === ERROR_CODES.NO_VALID_PATHS_EXTRACTED) {
+        response.push({
+          getData: () => {
+            const newData = getPagesWithIssues(auditData);
+            if (newData.length === 0) {
+              newData.push({
+                sourceSitemapUrl: reason.value,
+                pageUrl: '',
+                statusCode: 404,
+              });
+            }
+            return newData;
+          },
+          type: 'sitemap-no-valid-paths-extracted',
+          buildKey: (issue) => buildKeyOptionalPage(auditData, issue, 'sitemap-no-valid-paths-extracted'),
+          title: 'Sitemap issues found',
+          description: 'A complete and accurate sitemap helps search engines efficiently crawl and index your website pages,ensuring better visibility in search results. Fixing sitemap issues can improve the discoverability of your content.',
+        });
+      }
+    });
+  } else {
+    const newData = getPagesWithIssues(auditData);
+
+    if (newData.length) {
+      // some pages failed, create an opportunity when newData are all failed pages
+      response.push({
+        getData: () => newData,
+        type: 'pages-in-sitemap-have-errors',
+        buildKey: (issue) => buildKeyOptionalPage(auditData, issue, 'pages-in-sitemap-have-errors'),
+        title: 'Issues regarding pages in sitemaps',
+        description: 'All pages from sitemap should be available, with no redirects in place.',
+      });
+    }
+  }
+
+  return response;
+}
+
+async function createOpportunity(dataAccess, type, title, log, siteId, auditId) {
+  const opportunityData = {
+    siteId,
+    auditId,
+    type,
+    title,
+    runbook: 'https://adobe.sharepoint.com/:w:/r/sites/aemsites-engineering/Shared%20Documents/3%20-%20Experience%20Success/SpaceCat/Runbooks/Experience_Success_Studio_Sitemap_Runbook.docx?d=w6e82533ac43841949e64d73d6809dff3&csf=1&web=1&e=FTWy7t',
+    guidance: {
+      steps: [
+        'Verify each URL in the sitemap, identifying any that do not return a 200 (OK) status code.',
+        'Check RUM data to identify any sitemap pages with unresolved 3xx, 4xx or 5xx status codes – it should be none of them.',
+      ],
+    },
+    tags: ['Traffic Acquisition', 'Sitemap'],
```

Comment:
Can you remove "Sitemap" from here, as that is being already hardcoded on UI and other handlers are also not sending type tag

Reply:
Okey, done